src/phoenix/db/insertion/evaluation.py

```diff
@@ -73,9 +74,17 @@ async def _insert_trace_evaluation(
     ).where(models.Trace.trace_id == trace_id)
     if not (row := (await session.execute(stmt)).first()):
         raise InsertEvaluationError(
-            f"Cannot insert a trace evaluation for a missing trace: {trace_id=}"
+            f"Cannot insert a trace evaluation for a missing trace: {evaluation_name=}, {trace_id=}"
         )
     project_rowid, trace_rowid = row
+    if await session.scalar(
+        select(count())
+        .where(models.TraceAnnotation.name == evaluation_name)
+        .where(models.TraceAnnotation.trace_rowid == trace_rowid)
+    ):
+        raise InsertEvaluationError(
+            f"Cannot insert duplicate trace evaluation for: {evaluation_name=}, {trace_id=}"
+        )
```

Comment:
I think we want a unique constraint for evaluations, but we want to let them update, no?

Reply:
You're right. My oversight.

Reply:
This was why we had `sqlite_on_conflict="REPLACE"` on these indexes previously.